zonal_datacube/zonal_datacube.py

```diff
@@ -0,0 +1,262 @@
+from datetime import datetime
+from typing import Dict, List, Union, cast
+
+import dask.dataframe as dd
+import geopandas as gpd
+import pandas as pd
+import pystac
+import rasterio
+from odc import stac
+from rasterio import features
+from shapely import wkt
+
+from .analysis_functions import (
+    AnalysisFunction,
+    combine_agg_dicts,
+    get_default_analysis_function,
+)
+from .fishnet import fishnet
+
+
+class ZonalDataCube:
+    """A Dask-backed, lazy-loaded raster datacube that will only be read and
+    analyzed in specified zones."""
+
+    # Random time to set all STAC items to when spatial_only
+    # The time dimension will be removed after reading into xarray
+    DEFAULT_TIME = datetime.fromisoformat("2000-01-01T00:00:00")
+
+    def __init__(
+        self,
+        zones: gpd.GeoDataFrame,
+        stac_items: List[pystac.Item],
+    ):
+        """Creates a new zonal datacube. The datacube is lazy-loaded using the
+        STAC item specifications, and only portions of the datacube that fall
+        in the zones will be loaded at analysis time.
+
+        :param zones:
+            A GeoPandas DataFrame with a geometry column containing polygonal zones,
+            as well as any other attributes related to the zones.
+        :param stac_items:
+            A list of STAC Items
+            These will be read into a XArray dataset where each item is a data variable.
+        """
+
+        # allow this to be passed as a param once we support
+        # time series
+        self.spatial_only = True
+
+        self.zones = zones
+        self.stac_items = self._get_stac_items(stac_items, self.spatial_only)
+        self.attribute_columns = self._get_attribute_columns()
+
+        self.bounds = (0, 0, 10, 10)  # self.zones.total_bounds
```

Comment:
Are these absolute or relative bounds? If absolute, perhaps this needs to be passed-in as an arg instead of hard coded?

Reply:
oh good catch, that was definitely just for testing something